src/main/java/github/paz/awardportal/model/User/User.java

```diff
@@ -85,4 +89,51 @@ public void updateUser(BaseUser update){
 
     public User() {
     }
+
+
+    @Override
+    public String toString() {
+        return "User{" +
+                "id=" + id +
+                ", firstName='" + firstName + '\'' +
+                ", lastName='" + lastName + '\'' +
+                ", email='" + email + '\'' +
+                ", password='" + password + '\'' +
+                ", office=" + office +
+                ", isAdmin=" + isAdmin +
+                ", signature=" + Arrays.toString(signature) +
+                ", receivedAwards=" + receivedAwards +
+                '}';
+    }
+
+    // set signature from Multipartfile format which supports upload using <form>
+    public void setSignature(MultipartFile file) {
+        try {
+            this.signature = file.getBytes();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // return the signature in base64 string, allow it to be displayed in html
+    public String getEncodedSignature() {
+        try {
```

Comment:
Java 8 introduced standard library support for this, which could simplify the code.

String encodedSignature = Base64.getEncoder().encodeToString(signature);

Just pointing this out, this way works perfectly fine too.

Reply:
import java.util.Base64

is the import statement